### PR TITLE
Add suspend handler with new telemetry event for UWP scenarios

### PR DESCRIPTION
**Pull request by ryanlai2**
UWP apps don't terminate gracefully and must be caught in WinML during the suspend state to log telemetry.

This change adds a new telemetry event called "WinMLSuspended" which logs the sessionId

This change also removes dependency of WinML on cppwinrt in SDK.

### CODE REVIEW

**Review by zhangxiang1993 — APPROVED** [review]
:shipit:

**Review by angsabrina — APPROVED** [review]

File: cmake/nuget_helpers.cmake [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    get_filename_component(PACKAGES_DIR ${CMAKE_CURRENT_BINARY_DIR}/../packages ABSOLUTE)
+
+    # Restore nuget packages
+    add_custom_command(
+    OUTPUT ${target_dependency}
+    DEPENDS ${PACKAGES_CONFIG} ${NUGET_CONFIG}
+    COMMAND ${CMAKE_CURRENT_BINARY_DIR}/nuget/src/nuget restore ${PACKAGES_CONFIG} -PackagesDirectory ${PACKAGES_DIR} -ConfigFile ${NUGET_CONFIG}
+    VERBATIM)
+
+    add_custom_target(${nuget_target} ALL DEPENDS ${target_dependency})
```
Comment (tiagoshibata) [inline cmake/nuget_helpers.cmake:55]: Why add it to all?

**Review by tiagoshibata — DISMISSED** [review]

File: cmake/nuget_helpers.cmake (hunk 1 above)
Comment (ryanlai2) [reply cmake/nuget_helpers.cmake:55]: Yeah good point, only WinML needs to use it right now so should I remove it?

File: winml/lib/Api.Ort/OnnxruntimeEnvironment.cpp [hunk 2]
```diff
@@ -190,6 +191,18 @@ static void __stdcall WinmlOrtProfileEventCallback(const OrtProfilerEventRecord*
...
 }
 
+static void OnSuspending(winrt::Windows::Foundation::IInspectable const& sender, winrt::Windows::ApplicationModel::SuspendingEventArgs const& args) {
+  telemetry_helper.LogWinMLSuspended();
+}
+
+void OnnxruntimeEnvironment::RegisterSuspendHandler() {
+  try {
+    auto suspend_event_handler = winrt::Windows::Foundation::EventHandler<winrt::Windows::ApplicationModel::SuspendingEventArgs>(&OnSuspending);
+    suspend_token_ = winrt::Windows::ApplicationModel::Core::CoreApplication::Suspending(suspend_event_handler);
```
Comment (smk2007) [inline winml/lib/Api.Ort/OnnxruntimeEnvironment.cpp:201]: This should be OK, but did you verify if this pulls in any unwanted dependencies win 8.1?

**Review by tiagoshibata — APPROVED** [review]

File: winml/lib/Api.Ort/OnnxruntimeEnvironment.cpp (hunk 2 above)
Comment (ryanlai2) [reply winml/lib/Api.Ort/OnnxruntimeEnvironment.cpp:201]: Verified that the import dlls are the same before and after this change.

**Pull request merged**